src/Validator/ImageUrlConstraintValidator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Validator;
+
+use Normalizer;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\UrlValidator as UrlConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class ImageUrlConstraintValidator
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Validator
+ */
+class ImageUrlConstraintValidator extends UrlConstraintValidator {
+
+	/**
+	 * Checks if the passed value is valid.
+	 *
+	 * @param string     $value The value that should be validated
+	 * @param Constraint $constraint
+	 *
+	 * @throws UnexpectedTypeException If invalid constraint provided.
+	 * @throws UnexpectedValueException If invalid value provided.
+	 */
+	public function validate( $value, Constraint $constraint ) {
+		if ( ! $constraint instanceof ImageUrlConstraint ) {
+			throw new UnexpectedTypeException( $constraint, ImageUrlConstraint::class );
+		}
+
+		if ( null === $value || '' === $value ) {
+			return;
+		}
```

Comment:
❓ Pretty minor. But not sure 100% about this chunk of code. Seems for me that it's covered in L39 and L43?

Reply:
These lines are ignoring the validation rule and returns early so it can move on to the next validation rule (we can have multiple validation rules per attribute). In this case that's important because for the main image we also check the Constraint NotBlank, whereas we don't want that constraint to apply for gallery images.

The validator docs explain that a bit clearer: https://symfony.com/doc/current/validation/custom_constraint.html#creating-the-validator-itself